simplq/src/components/pages/Status/QueueDetails.jsx

```diff
@@ -1,31 +1,29 @@
-import React, { useState, useEffect } from 'react';
+import React, { useEffect, useCallback } from 'react';
 import InfoIcon from '@material-ui/icons/Info';
-import QueueStats from '../../common/QueueStats';
-import { QueueRequestFactory } from '../../../api/requestFactory';
+import QueueStats from 'components/common/QueueStats';
+import { useGetQueueStatus } from 'store/asyncActions';
+import { selectQueueStatus } from 'store/queueStatus';
+import { useDispatch, useSelector } from 'react-redux';
 import SidePanelItem from '../../common/SidePanel/SidePanelItem';
-import useRequest from '../../../api/useRequest';
 
-export default (props) => {
-  const [queueStatusResponse, setQueueStatusResponse] = useState();
-  const { requestMaker } = useRequest();
+export default ({ queueId }) => {
+  const queueStatus = useSelector(selectQueueStatus);
+  const dispatch = useDispatch();
+  const getQueueStatus = useCallback(useGetQueueStatus(), []);
 
   useEffect(() => {
-    async function fetchData() {
-      const response = await requestMaker(QueueRequestFactory.getStatus(props.queueId));
-      setQueueStatusResponse(response);
-    }
-    fetchData();
-  }, [props.queueId, requestMaker]);
+    dispatch(getQueueStatus({ queueId }));
+  }, [queueId, dispatch, getQueueStatus]);
 
   return (
     <SidePanelItem
       Icon={InfoIcon}
       title="Queue Details"
       description="Other information about the queue"
       expandable
-      loading={!queueStatusResponse}
+      loading={!queueStatus}
     >
-      <QueueStats queueStatus={queueStatusResponse} />
+      <QueueStats queueStatus={queueStatus} />
```

Comment:
This won't work, we would have to check for empty object. 🤔 

An isLoading flag might be better, as you were implementing before?

These kind of glitches we can fix in a follow up PR as well.

Reply:
All `pending', `fulfilled`, and `rejected` should be handled in reducers. It can be done later.

Reply:
code looks much cleaner now